pkg/api/globalquery/server/server_test.go

```diff
@@ -0,0 +1,16 @@
+package server
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateOpenAPISpec(t *testing.T) {
+	buf := &bytes.Buffer{}
+	s := New("localhost:8146", nil, nil)
```

Comment:
Tests should IMHO try to avoid actually firing up servers wherever possible (there's no guarantee that a) you're allowed to do that on any node that runs these tests and b) the port is available, which can lead to weird / totally random failures). Can't you just use the underlying handlers directly or in memory without actually firing up the server?

Reply:
Don't worry. The server isn't started up until you call `Start`.